### PR TITLE
Add callback for successful connection to a conference

### PR DESCRIPTION
**Pull request by zugz**
(based on #1035)

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1042)

### CODE REVIEW

File: toxcore/tox.api.h [hunk 1]
```diff
@@ -2135,6 +2135,17 @@ namespace conference {
   }
 
 
+  /**
+   * This event is triggered when the client successfully connects to a conference.
```
Comment (iphydf) [inline toxcore/tox.api.h]: Can you document somewhere what guarantees this gives the client? Perhaps in the docs for "invite", explain in what state it's valid to call "invite", and when it stops being valid.

File: toxcore/group.c [hunk 2]
```diff
@@ -1193,6 +1193,15 @@ void g_callback_group_invite(Group_Chats *g_c, g_conference_invite_cb *function)
     g_c->invite_callback = function;
 }
 
+/* Set the callback for group connection.
+ *
+ *  Function(Group_Chats *g_c, void *userdata)
```
Comment (iphydf) [inline toxcore/group.c]: Let's remove this `Function(...)` stuff. It's documentation that's easy to get out of date and it's not really useful.

**Review by iphydf — CHANGES_REQUESTED** [review]

File: toxcore/tox.c [hunk 3]
```diff
@@ -1126,6 +1126,12 @@ void tox_callback_conference_invite(Tox *tox, tox_conference_invite_cb *callback
     g_callback_group_invite((Group_Chats *)m->conferences_object, (g_conference_invite_cb *)callback);
 }
 
+void tox_callback_conference_connected(Tox *tox, tox_conference_connected_cb *callback)
+{
+    Messenger *m = tox;
```
Comment (iphydf) [inline toxcore/tox.c]: This is now different, since we have a `struct Tox`.

**Comment by zugz** [comment]
>Perhaps in the docs for "invite", explain in what state it's valid to 
>call "invite", and when it stops being valid.

Done

File: toxcore/group.c (hunk 2 above)
Comment (zugz) [reply toxcore/group.c]: Done

**Review by iphydf — APPROVED** [review]

**Comment by iphydf** [comment]
@zugz please rebase/squash, then merge.

**Pull request merged**